### PR TITLE
chore: Add DELL XPS 9510

### PR DESCRIPTION
**Pull request by ktaf**
###### Description of changes

Add DELL XPS 9510

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -50,6 +50,8 @@
       dell-xps-15-7590-nvidia = import ./dell/xps/15-7590/nvidia;
       dell-xps-15-9500 = import ./dell/xps/15-9500;
       dell-xps-15-9500-nvidia = import ./dell/xps/15-9500/nvidia;
+      dell-xps-15-9510 = import ./dell/xps/15-9510;
```
Comment (Mic92) [inline flake.nix:53]: Please also add this to the README table where our CI picks it up for testing.

Comment (ktaf) [reply flake.nix:53]: Done! Thanks for letting me know @Mic92 \m/

**Comment by Mic92** [comment]
@mergify rebase

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**